addons/survey/controllers/main.py

```diff
@@ -467,10 +467,7 @@ def survey_submit(self, survey_token, answer_token, **post):
 
         errors = {}
         # Prepare answers / comment by question, validate and save answers
-        inactive_questions = request.env['survey.question'] if answer_sudo.is_session_answer else answer_sudo._get_inactive_conditional_questions()
         for question in questions:
-            if question in inactive_questions:  # if question is inactive, skip validation and save
-                continue
```

Comment:
I think this should be done like that instead : 
```
for question in questions:            
    inactive_questions = request.env['survey.question'] if answer_sudo.is_session_answer else answer_sudo._get_inactive_conditional_questions()    
    if question in inactive_questions:  # if question is inactive, skip validation and save                
        continue
```
The idea is to re-evaluate which question is active or not, after submitting question per question. In page per question, this is not mandatory as only one question is submitted one by one. For the two other cases, as multiple questions are submitted at the same time, we cannot know before saving the question if the following will still be active or not.

removing the check completely is dangerous as we assume that the frontend is doing all the job of filtering inactive questions before submitting - which is the case - but by security, we should align both behaviour at frontend and backend side.